### PR TITLE
Move Calypso redirect option to Jetpack options

### PR DESCRIPTION
**Pull request by vindl**
Instead of using a separate site option for Calypso redirect switch, let's utilize existing Jetpack options functionality. This was suggested by @beaulebens in https://github.com/Automattic/jetpack/pull/7702.

#### Testing instructions

1. Use a Jetpack connected test site.
2. Since the option will not be set, change [this line](https://github.com/Automattic/jetpack/compare/update/move-calypso-edit-option?expand=1#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R618) to:
`if ( Jetpack::get_option( 'edit_links_calypso_redirect', true ) && ! is_admin() ) {`
3. Open a post or page from your site's front end and verify that edit links redirect correctly to Calypso.
Test with supported Custom Post Types too (Portfolios and Testimonials).

### CODE REVIEW

**Comment by dereksmart** [comment]
So this won't automatically delete it on deactivation nor deletion.  We are pretty careful about what data we delete on deactivation, since we very often suggest either deactivation/reinstall of Jetpack as a debug process.  Actually, currently we don't really clean up anything on deactivation, other than the connection options, and we clean the same options on disconnect from wp.com.  Check the `[plugin_deactivation](plugin_deactivation)` method, which is hooked to `register_deactivation_hook()`.

**Comment by jeremysawesome** [comment]
Confirmed the edit redirect using the Testing Instructions. The code itself looks good to me 👍  

It sounds like we might need to do a bit more to include this option in the plugin cleanup?

Also, I'm not sure if we want to look into the `Reset Options` beta link that @beaulebens mentioned in his comment here: https://github.com/Automattic/jetpack/pull/7702#issuecomment-325835079

**Comment by vindl** [comment]
> So this won't automatically delete it on deactivation nor deletion. We are pretty careful about what data we delete on deactivation, since we very often suggest either deactivation/reinstall of Jetpack as a debug process, and we don't want users losing settings while trying to fix something.
> 
> Actually, currently we don't really clean up anything on deactivation, other than the connection options, and we clean the same options on disconnect from wp.com. Check the [plugin_deactivation] method, which is hooked to register_deactivation_hook().

Thanks for the clarification @dereksmart, I updated the PR summary to account for that. I somehow misunderstood that part from the original comment and assumed that cleanup would be automatic. 

If that is the case, I don't know why should we make an exception for this option and clean it up? I don't see how it's different when compared to other Jetpack settings/options and I think it should be handled in the same way.

**Comment by vindl** [comment]
> It sounds like we might need to do a bit more to include this option in the plugin cleanup?

@jeremysawesome I don't think it should be included in the plugin cleanup (see my previous comment).

> Also, I'm not sure if we want to look into the Reset Options beta link that @beaulebens mentioned in his comment here: #7702 (comment)

It doesn't look like we need to do anything else in order for this option to be picked up by `Reset Options`. For more info you can check out [this function](https://github.com/Automattic/jetpack/blob/30a3f000bfe81ca463daa0c5e24a5de4d0902d58/_inc/lib/class.core-rest-api-endpoints.php#L835).

In conclusion, the changes in this PR look sufficient to me, but please let me know if y'all don't agree.

**Comment by dereksmart** [comment]
> I don't know why should we make an exception for this option and clean it up?

I can't think of any reason why we'd want to either!  I wanted to make it clear that it wasn't going to, since it sounded like that was a goal judging by the original PR comment, but this change as it it LGTM!

**Comment by dereksmart** [comment]
pushed to 5.3 48e387bb4
